site/src/xServices/createTemplate/createTemplateXService.ts

```diff
@@ -0,0 +1,393 @@
+                onDone: {
+                  target: "waitingForJobToBeCompleted",
+                  actions: ["assignVersion"],
+                },
+                onError: {
+                  actions: ["assignError"],
+                  target: "#createTemplate.idle",
+                },
+              },
+              tags: ["submitting"],
+            },
+            waitingForJobToBeCompleted: {
+              invoke: {
+                src: "waitForJobToBeCompleted",
+                onDone: [
+                  {
+                    target: "loadingMissingParameters",
+                    cond: "hasMissingParameters",
+                    actions: ["assignVersion"],
+                  },
+                  {
+                    target: "#createTemplate.idle",
+                    actions: ["displayJobError"],
+                    cond: "hasFailed",
+                  },
+                  { target: "creatingTemplate", actions: ["assignVersion"] },
+                ],
+                onError: {
+                  target: "#createTemplate.idle",
+                  actions: ["assignError"],
+                },
+              },
+              tags: ["submitting"],
+            },
+            loadingMissingParameters: {
+              invoke: {
+                src: "loadParameterSchema",
+                onDone: {
+                  target: "promptParameters",
+                  actions: ["assignParameters"],
+                },
+                onError: {
+                  target: "#createTemplate.idle",
+                  actions: ["assignError"],
+                },
+              },
+              tags: ["submitting"],
+            },
+            promptParameters: {
+              on: {
+                CREATE: {
+                  target: "creatingVersionWithParameters",
+                  actions: ["assignTemplateData"],
+                },
+              },
+            },
+            creatingVersionWithParameters: {
+              invoke: {
+                src: "createVersionWithParameters",
+                onDone: {
+                  target: "waitingForJobToBeCompleted",
+                  actions: ["assignVersion"],
+                },
+                onError: {
+                  actions: ["assignError"],
+                  target: "promptParameters",
+                },
+              },
+              tags: ["submitting"],
+            },
+            creatingTemplate: {
+              invoke: {
+                src: "createTemplate",
+                onDone: {
+                  target: "created",
+                  actions: ["onCreate"],
+                },
+                onError: {
+                  actions: ["assignError"],
+                  target: "#createTemplate.idle",
+                },
+              },
+              tags: ["submitting"],
+            },
+            created: {
+              type: "final",
+            },
+          },
+        },
+      },
+    },
+    {
+      services: {
```

Comment:
I would probably break these services into a separate file just to minimize scrolling. I find myself scrolling a lot when trying to understand XState machines :)

Reply:
Yes. Unfortunately XState typegen does not work pretty well with that but I think I can create separated services tho. Let's see if it gets better.